### DISCUSSION TITLE
Problemi con integrazione Shelly

### DISCUSSION THREAD

**Discussion by riccardoiacobucci** [indomus/forum#2324] [HUB: Home Assistant]
Sono entrato di recente nel mondo di HA, ho già configurato diversi dispositivi ma sto incontrando problemi con gli Shelly. Al momento di creare l'integrazione ricevo il seguente messaggio: "Prima della configurazione, i dispositivi alimentati a batteria devono essere riattivati, ora è possibile riattivare il dispositivo utilizzando un pulsante su di esso.".
Sotto il messaggio, la richiesta di inserire un "Host" come campo obbligatorio.

Ho diversi Shelly 2PM, 1PM e Dimmer, ma nessuno di questi viene rilevato e nessuno è a batteria. Ho anche cercato in rete, ma non sono riuscito a trovare nessuno che abbia avuto lo stesso problema, nè spunti tra la documentazione.
Ho tentato ad inserire qualche ip degli shelly, quello di HA, ma me li da tutti non validi e il messaggio non mi manda avanti.

Ho anche tentato di risolvere utilizzando Shellyforhass, ma anche lì nessun dispositivo viene rilevato.
Via MQTT sono riuscito ad integrarne uno, solo parzialmente, ma ho visto che la procedura è piuttosto per il mio livello attuale.

Qualcuno potrebbe aiutarmi ad uscirne fuori?

Grazie in anticipo
Riccardo

![Immagine 2023-02-03 132024](https://user-images.githubusercontent.com/124188529/217032291-9708cec0-d71a-416f-8d6c-f930cd84eeea.jpg)

**Comment by snakuzzo** [comment]
Dimenticati di shellyforhass, se sono device di generazione 1, abbiamo quasi tutto sul sito per fare tutte le configurazioni del caso
Non hai specificato che tipo di installazione hai

Se vuoi usare l'integrazione nativa leggi la [documentazione ufficiale](https://www.home-assistant.io/integrations/shelly/#shelly-device-configuration-generation-1)

**Reply by riccardoiacobucci** [reply]
Hai ragione... ho installato HA Core su docker. Per ora su windows 10, poi valuterò magari migrazione su minipc.
Gli shellies sono tutti di seconda generazione plus 2PM, plus 1pm e dimmer2.

Provo a ricominciare da capo seguendo alla lettera la documentazione, ma quando avevo provato sempre al messaggio bloccante dell'host mi ero fermato.

Grazie
Riccardo

**Reply by snakuzzo** [reply]
controlla i log di core e se hai qualche regola sul firewall di Windows.
Io ho provato qualche giorno fa su Windows e funzionava tutto senza alcun problema.

**Reply by riccardoiacobucci** [reply]
niente da fare...
nei log di HA non viene registrato nulla, nè quando tento di aggiungere l'integrazione e nè quando poi tento di inserire il valore di un host e il frontend mi risponde che è impossibile connettersi.
Lato PC ho proprio disattivato il firewall windows, ma non ci sono differenze.
Non dovrebbe neanche dipendere dalla versione di HA, perché ho installato l'ultima (23.2.1) e me lo dava anche prima.

**Reply by snakuzzo** [reply]
Non so che dirti, mi spiace.
Come già ti ho detto ieri, ho fatto un test qualche giorno fa proprio da Windows e non ho riscontrato alcun problema.
Fai una prova...
Apri una shell del container...

```
docker exec -it homeassistant bash
```

e prova un ping...

```
ping <ipshelly>
```

Lo raggiungi ?

**Reply by riccardoiacobucci** [reply]
Pare sia ok. Ieri sono anche riuscito a farne accendere uno con mqtt, ma ho visto che il resto della configurazione era abbastanza complessa e vorrei far funzionare l'integrazione nativa.

35 packets transmitted, 35 packets received, 0% packet loss
round-trip min/avg/max = 4.513/12.052/35.872 ms

**Reply by snakuzzo** [reply]
Controlla bene la configurazione [QUI](https://www.home-assistant.io/integrations/shelly/#shelly-device-configuration-generation-1).
Assicurati di configurare come indicato la parte relativa a CoIoT e, se puoi, ricrea il container di HA esponendo anche la porta 5683.

Fatto questo riprova. Fai sapere come va

**Reply by riccardoiacobucci** [reply]
Ho ricaricato il container e adesso è in ascolto anche sulla 5683. Però sulle impostazione dello shelly non ho CoIoT. Sono seconda gen e ho il Websocket che ho configurato (solo su uno degli 8) con il server ws://192.168.178.45:8123/api/shelly/ws come da documentazione. Posso scegliere il tipo connessione (TLS no validation, default TLS, user TLS) ma non mi pare cambi nulla tra una e l'altra.

**Reply by riccardoiacobucci** [reply]
Sull'unico dimmer ho visto che le impostazioni hanno il CoIoT. Era abilitato e l'ho configurato come da istruzioni, sulla porta 5683.
Il problema però è rimasto

**Reply by snakuzzo** [reply]
Ho uno shelly1 plus (quindi 2nd gen). Appena posso provo anche io e ti faccio sapere se salta fuori qualcosa. Il test lo avevo fatto con uno shelly1 (1st gen).

**Reply by riccardoiacobucci** [reply]
Aggiungo un'altra informazione. 
Ho provato anche a scollegare il dimmer2, visto che era l'unico con il protocollo CoIoT. Il problema resta anche con solo Shelly 2PM plus e 1PM plus. Senza firewall windows sul pc. Mi sono spulciato pure le varie opzioni del router (Fritzbox 7490) ma non mi pare di vedere nulla che possa creare problemi.

**Reply by snakuzzo** [reply]
Ok...ti aggiorno appena riesco a fare un test
